dev.py

```diff
@@ -763,6 +796,32 @@ def scipy_tests(cls, args, pytest_args):
                   f"installed at:{mod_path}")
             # runner verbosity - convert bool to int
             verbose = int(args.verbose) + 1
+
+            scipy_code_fetching_gcov_flag = (
+                "import scipy;"
+                "print(scipy.show_config(mode='dicts')['Coverage Information']['GCOV'])")
+            was_built_with_gcov_flag = subprocess.run(["python", "-c",
+                                scipy_code_fetching_gcov_flag.format(lang="'c++'")],
+                                capture_output=True, encoding='utf-8').stdout.strip()
+            if was_built_with_gcov_flag == "true":
+                scipy_code_for_checking_compiler = (
+                    "import scipy;"
+                    "print(scipy.show_config(mode='dicts')['Compilers'][{lang}]['name'])")
+                cpp = subprocess.run(["python", "-c",
+                                      scipy_code_for_checking_compiler.format(lang="'c++'")],
+                                      capture_output=True, encoding='utf-8').stdout.strip()
+                c = subprocess.run(["python", "-c",
+                                    scipy_code_for_checking_compiler.format(lang="'c'")],
+                                    capture_output=True, encoding='utf-8').stdout.strip()
+                fortran = subprocess.run(["python", "-c",
+                                          scipy_code_for_checking_compiler.format(lang="'fortran'")],
+                                          capture_output=True, encoding='utf-8').stdout.strip()
+                if not (c == 'gcc' and cpp == 'gcc' and fortran == 'gcc'):
+                    print("SciPy was built with --gcov flag which requires LCOV while running tests")
+                    print("Further, LCOV usage requires GCC for C, C++ and Fortran codes in SciPy")
+                    print(f"Compilers used currently are C: {c}, C++: {cpp}, Fortran: {fortran}")
+                    print("Therefore, exiting without running tests")
+                    exit(1) # Exit because tests won't succeed and will give missing symbol error
```

Comment:
There are two parts to this change,

1. Checking for compilers used to build SciPy - Reason for this is that the failure in https://github.com/scipy/scipy/issues/15896#issuecomment-2485999246 is not linked to the OS but it is linked to the compiler being used. I built SciPy using `clang-14` on my Linux machine with `--gcov` option. It generated the same error i.e., the following,

Traceback on Linux when SciPy built using `clang-14` with `--gcov` option:

<details>

```zsh
_ ERROR collecting build-install/lib/python3.12/site-packages/scipy/stats/tests/test_tukeylambda_stats.py _
ImportError while importing test module '/home/czgdp18079/Quansight/scipy/build-install/lib/python3.12/site-packages/scipy/stats/tests/test_tukeylambda_stats.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/home/czgdp18079/miniforge3/envs/scipy-dev/lib/python3.12/importlib/__init__.py:90: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
scipy/stats/__init__.py:617: in <module>
    from ._stats_py import *
scipy/stats/_stats_py.py:38: in <module>
    from scipy import sparse
scipy/__init__.py:136: in __getattr__
    return _importlib.import_module(f'scipy.{name}')
/home/czgdp18079/miniforge3/envs/scipy-dev/lib/python3.12/importlib/__init__.py:90: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
scipy/sparse/__init__.py:312: in <module>
    from . import csgraph
scipy/sparse/csgraph/__init__.py:187: in <module>
    from ._laplacian import laplacian
scipy/sparse/csgraph/_laplacian.py:7: in <module>
    from scipy.sparse.linalg import LinearOperator
scipy/sparse/linalg/__init__.py:132: in <module>
    from ._eigen import *
scipy/sparse/linalg/_eigen/__init__.py:9: in <module>
    from .arpack import *
scipy/sparse/linalg/_eigen/arpack/__init__.py:20: in <module>
    from .arpack import *
scipy/sparse/linalg/_eigen/arpack/arpack.py:50: in <module>
    from . import _arpack
E   ImportError: /home/czgdp18079/Quansight/scipy/build-install/lib/python3.12/site-packages/scipy/sparse/linalg/_eigen/arpack/_arpack.cpython-312-x86_64-linux-gnu.so: undefined symbol: llvm_gcda_summary_info
_ ERROR collecting build-install/lib/python3.12/site-packages/scipy/stats/tests/test_variation.py _
ImportError while importing test module '/home/czgdp18079/Quansight/scipy/build-install/lib/python3.12/site-packages/scipy/stats/tests/test_variation.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/home/czgdp18079/miniforge3/envs/scipy-dev/lib/python3.12/importlib/__init__.py:90: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
scipy/stats/__init__.py:617: in <module>
    from ._stats_py import *
scipy/stats/_stats_py.py:38: in <module>
    from scipy import sparse
scipy/__init__.py:136: in __getattr__
    return _importlib.import_module(f'scipy.{name}')
/home/czgdp18079/miniforge3/envs/scipy-dev/lib/python3.12/importlib/__init__.py:90: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
scipy/sparse/__init__.py:312: in <module>
    from . import csgraph
scipy/sparse/csgraph/__init__.py:187: in <module>
    from ._laplacian import laplacian
scipy/sparse/csgraph/_laplacian.py:7: in <module>
    from scipy.sparse.linalg import LinearOperator
scipy/sparse/linalg/__init__.py:132: in <module>
    from ._eigen import *
scipy/sparse/linalg/_eigen/__init__.py:9: in <module>
    from .arpack import *
scipy/sparse/linalg/_eigen/arpack/__init__.py:20: in <module>
    from .arpack import *
scipy/sparse/linalg/_eigen/arpack/arpack.py:50: in <module>
    from . import _arpack
E   ImportError: /home/czgdp18079/Quansight/scipy/build-install/lib/python3.12/site-packages/scipy/sparse/linalg/_eigen/arpack/_arpack.cpython-312-x86_64-linux-gnu.so: undefined symbol: llvm_gcda_summary_info
```

</details>

Without `--gcov` option, all the tests work fine with `clang-14` build on SciPy.

So that's why I had to check for compilers and not the OS to generate error message and exit without running tests.

2. Checking for usage of `--gcov` option during SciPy build - As I explained in point 1, we need to exit without running tests when SciPy is built using `clang` and the `--gcov` option. So, I save GCOV option value in `scipy/__config__.py.in` and then fetch it here. If it's `"true"` only then we perform further checks for compilers. Otherwise everything should be fine with `clang`.